src/Engine/ProtoCore/DSASM/InstructionSet.cs

```diff
@@ -693,25 +937,35 @@ public static List<StackValue> BuildInvalidRegisters()
         /// type of StackValue should be AddressType.ArrayKey. 
         /// </summary>
         /// <param name="array"></param>
-        /// <param name="index"></param>
+        /// <param name="key"></param>
         /// <returns></returns>
-        public bool TryGetArrayKey(out StackValue array, out int index)
+        public bool TryGetArrayKey(out StackValue array, out int key)
         {
             array = StackValue.Null;
-            index = Constants.kInvalidIndex;
+            key = Constants.kInvalidIndex;
 
-            if (!this.IsArrayKey || opdata == Constants.kInvalidIndex)
-            {
+            if (!IsArrayKey)
                 return false;
```

Comment:
If you do not want to put a pair of curly brackets (as per coding standard), then it can be written as `if (!IsArrayKey) return false;`